ocaml/xapi/xapi_clustering.ml

```diff
@@ -154,6 +154,24 @@ let assert_operation_host_target_is_localhost ~__context ~host =
   if host <> Helpers.get_localhost ~__context then
     raise Api_errors.(Server_error (internal_error, [ "A clustering operation was attempted from the wrong host" ]))
 
+(* a new member of a cluster should reside on the same network as the cluster *)
+let assert_same_subnet ~__context ~cluster ~pIF =
+  let parse_ip ip_str =
+    match Ipaddr.of_string ip_str with
+    | Ok ip_addr       -> ip_addr
+    (* this shouldn't happen because we shouldn't be storing invalid ips *)
+    | Error (`Msg msg) -> raise Api_errors.(Server_error (internal_error, [Printf.sprintf "could not parse ip address (%s), msg: %s" ip_str msg]))
```

Comment:
Something like this could make this a bit easier:

```
let internal_error fmt = 
  Printf.kprintf (fun msg -> raise Api_errors.(Server_error (internal_error, msg)))
...
internal_error "could not parse IP address %s: %s" ip_str msg